### PR TITLE
Issue #90: Refactored validation infrastructure and integrated package n...

### PR DESCRIPTION
**Pull request by bmuschko**
...ame validation.

### CODE REVIEW

**Comment by bmuschko** [comment]
Build is fixed now. Was a problem on JCenter.

File: src/test/groovy/com/netflix/gradle/plugins/deb/validation/DebPackageNameValidatorTest.groovy [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    def "verifies #description: '#attribute'"() {
+        when:
+        boolean valid = validator.validate(attribute)
+
+        then:
+        valid == result
+
+        where:
+        attribute            | result | description
+        'abc-1.0'            | true   | 'valid package name with mixed alphanumeric characters'
```
Comment (rspieldenner) [inline src/test/groovy/com/netflix/gradle/plugins/deb/validation/DebPackageNameValidatorTest.groovy]: This test case is slightly confusing to me, its unclear if the 1.0 is part of the package name or the version. Add a few more true examples.

File: src/test/groovy/com/netflix/gradle/plugins/rpm/validation/RpmPackageNameAttributeValidatorTest.groovy [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    def "verifies #description: '#attribute'"() {
+        when:
+        boolean valid = validator.validate(attribute)
+
+        then:
+        valid == result
+
+        where:
+        attribute       | result | description
+        'aBc_hello-1.0' | true   | 'valid package name with mixed alphanumeric characters'
```
Comment (rspieldenner) [inline src/test/groovy/com/netflix/gradle/plugins/rpm/validation/RpmPackageNameAttributeValidatorTest.groovy]: Could you add some more true examples? Possibly confusing on whether the 1.0 is a version or part of the package name.

File: src/test/groovy/com/netflix/gradle/plugins/deb/validation/DebPackageNameValidatorTest.groovy (hunk 1 above)
Comment (bmuschko) [inline src/test/groovy/com/netflix/gradle/plugins/deb/validation/DebPackageNameValidatorTest.groovy]: Based on the package name rules this is a valid package name. I was aiming more for covering all possible combinations.

File: src/test/groovy/com/netflix/gradle/plugins/rpm/validation/RpmPackageNameAttributeValidatorTest.groovy (hunk 2 above)
Comment (bmuschko) [inline src/test/groovy/com/netflix/gradle/plugins/rpm/validation/RpmPackageNameAttributeValidatorTest.groovy]: Based on the package name rules this is a valid package name. I was aiming more for covering all possible combinations.

**Comment by bmuschko** [comment]
I added some more test cases.

**Pull request merged**